examples/keyset.rs

```diff
@@ -0,0 +1,524 @@
+    if !args.is_empty() {
+        eprintln!("Usage: ... status");
+        exit(1);
+    }
+
+    let ks = load_keyset(filename);
+
+    println!("Keys:");
+    let keys = ks.keys();
+    for key in keys {
+        match key.keytype() {
+            KeyType::Ksk(keystate)
+            | KeyType::Zsk(keystate)
+            | KeyType::Include(keystate) => {
+                println!(
+                    "\t{} {}",
+                    key.pubref(),
+                    key.privref().unwrap_or_default(),
+                );
+                println!("\t\tState: {}", keystate);
+            }
+            KeyType::Csk(keystate_ksk, keystate_zsk) => {
+                println!(
+                    "\t{} {}",
+                    key.pubref(),
+                    key.privref().unwrap_or_default(),
+                );
+                println!("\t\tKSK role state: {}", keystate_ksk,);
+                println!("\t\tZSK role state: {}", keystate_zsk,);
+            }
+        }
+        let ts = key.timestamps();
+        println!(
+            "\t\tCreated: {}",
+            ts.creation()
+                .map_or("<empty>".to_string(), |x| x.to_string()),
+        );
+        println!(
+            "\t\tPublished: {}",
+            ts.published()
+                .map_or("<empty>".to_string(), |x| x.to_string())
+        );
+        println!(
+            "\t\tVisible: {}",
+            ts.visible()
+                .map_or("<empty>".to_string(), |x| x.to_string()),
+        );
+        println!(
+            "\t\tDS visible: {}",
+            ts.ds_visible()
+                .map_or("<empty>".to_string(), |x| x.to_string())
+        );
+        println!(
+            "\t\tRRSIG visible: {}",
+            ts.rrsig_visible()
+                .map_or("<empty>".to_string(), |x| x.to_string()),
+        );
+        println!(
+            "\t\tWithdrawn: {}",
+            ts.withdrawn()
+                .map_or("<empty>".to_string(), |x| x.to_string())
+        );
+    }
+    let rolls = ks.rollstates();
+    let active_rolls = rolls.keys();
+    if active_rolls.len() == 0 {
+        println!("No rolls in progress");
+    } else {
+        println!("Rolls:");
+        for roll in active_rolls {
+            println!("\t{roll:?}: {:?}", rolls.get(roll).unwrap());
+        }
+    }
+}
+
+fn save_keyset(filename: &str, ks: &KeySet) {
+    let json = serde_json::to_string(&ks).unwrap();
+
+    let mut file = File::create(filename).unwrap();
+    write!(file, "{json}").unwrap();
+}
+
+fn load_keyset(filename: &str) -> KeySet {
+    let file = File::open(filename).unwrap();
+    let ks: KeySet = serde_json::from_reader(file).unwrap();
+    ks
+}
+
+fn str_to_rolltype(rolltype: &str) -> RollType {
+    if rolltype == "ksk-roll" {
+        RollType::KskRoll
+    } else if rolltype == "zsk-roll" {
+        RollType::ZskRoll
+    } else if rolltype == "csk-roll" {
+        RollType::CskRoll
+    } else {
+        panic!("Unknown roll type {rolltype}");
+    }
+}
```

Comment:
Perhaps `RollType` should implement `FromStr` instead?

Reply:
Done